### ISSUE TITLE
Kal'Falnl C'ndros was allowed to fit on the Millenium Falcon

### ISSUE THREAD

**Issue by BlackbeardBen** [PlayersCommittee/gemp-swccg-public#221]
Just played a game where Kal'Falnl C'ndros was allowed to fit on the Millenium Falcon - a starfighter that she shouldn't be allowed aboard.  She slotted in as a passenger - I'm not sure if she could pilot or not.

**Comment by jdipaolo1** [comment]
@BlackbeardBen Do you have a replay link?

**Comment by BlackbeardBen** [comment]
> @BlackbeardBen Do you have a replay link?

Yeah, here: https://gemp.starwarsccg.org/gemp-swccg/game.html?replayId=BenKenobi$msnf774pak0bff3j

**Comment by BOTVHD** [comment]
This happened on LS turn 3.
- Nebulon-B Frigate (with KFC aboard as pilot) and Millenium Falcon ship-dock for 1 force
- KFC transfers from Nebulon-B Frigate to Millenium Falcon as a passenger
- Nebulon-B Frigate and Millenium Falcon (with KFC aboard) undock

Likely a lost modifier bug (MayNotExistAtTargetModifier on KFC), but should attempt to reproduce with ship docking?

**Comment by BOTVHD** [comment]
Was not able to replicate using a basic scripted test ( see KalFalnlCndrosCannotTransferToStarfighterUsingShipDock() ), but I did not replicate the entire board state.

Here is the board state from the replay, just after the transfer:
<img width="1919" height="966" alt="Image" src="https://github.com/user-attachments/assets/bc0895e1-3ea6-4b10-b6da-03f9eb2798ef" />

I don't see any locations with special deploy permission (like Trash Compactor) that would potentially incorrectly override KFC's no-starship modifier.

**Comment by billbisco** [comment]
Not a bug in current engine. Recommend no code change. Recommend close.

Printed Kal'Falnl C'ndros: may not deploy to or board starfighters or enclosed vehicles. Millennium Falcon is a starfighter, so she cannot deploy aboard it, embark onto it, or transfer onto it (including while two ships are docked).

That is implemented as may-not-deploy-to-target and may-not-exist-at-target on starfighters and enclosed vehicles. She can still go aboard a capital ship (Nebulon-B).

The report is a 2020 replay where she transferred from a docked Nebulon-B onto Falcon as a passenger. Tests covering that path (deploy to Falcon, embark on Falcon, and ship-dock transfer from Nebulon-B to Falcon) were run against current engine. The Falcon passenger transfer does not reproduce: after the ships dock, Transfer is offered for a normal character and not for Kal'Falnl.

That replay does not match current play. No engine change.

Recommend closing this issue.

**Comment by BOTVHD** [comment]
> Not a bug in current engine. Recommend no code change. Recommend close. 
... 
> That replay does not match current play. No engine change.
> 
> Recommend closing this issue.

I would clarify that to: we don't know how to reproduce that bug in the current code.

That could be because it was fixed by some other change in the last 6 years (and "not a bug in current engine" is actually true!).  Or it could be because we just don't know the complex conditions that lead to it.  Or it could be an example of a lost modifier issue.

I don't think closing it is the right approach until we know it was actually fixed.

If you have extra unused processing available, some day you might be able to have AI review the code base at the time this bug occurred (Nov 15, 2020 - maybe https://github.com/PlayersCommittee/gemp-swccg-public/tree/e89f7639aed6f332d3c9169fc76ec516859b3551 ) and identify that this was a legit, reproducible bug at the time.  Then, we can see what PR/commit fixed the issue and close this?
